proxyfinder.go

```diff
@@ -85,34 +89,36 @@ func (pf *ProxyFinder) findProxyForRequest(req *http.Request) (*url.URL, error)
 		log.Printf(`[%d] %s %s via "DIRECT" (not connected)`, id, req.Method, req.URL)
 		return nil, nil
 	}
-	s, err := pf.runner.FindProxyForURL(req.URL)
+	str, err := pf.runner.FindProxyForURL(req.URL)
 	if err != nil {
 		return nil, err
 	}
-	log.Printf("[%d] %s %s via %q", id, req.Method, req.URL, s)
-	ss := strings.Split(s, ";")
-	if len(ss) > 1 {
-		log.Printf("[%d] Warning: ignoring all but first proxy in %q", id, s)
-	}
-	trimmed := strings.TrimSpace(ss[0])
-	if trimmed == "DIRECT" {
-		return nil, nil
-	}
-	var host string
-	n, err := fmt.Sscanf(trimmed, "PROXY %s", &host)
-	if err == nil && n == 1 {
-		// The specified proxy should contain both a host and a port, but if for some reason
-		// it doesn't, assume port 80. This needs to be made explicit, as it eventually gets
-		// passed to net.Dial, which also requires a port.
-		proxy := &url.URL{Host: host}
-		if proxy.Port() == "" {
-			proxy.Host = net.JoinHostPort(host, "80")
+	arr := strings.Split(str, ";")
```

Comment:
technically, strings.Split returns a slice not array, so this name is not quite right. you could just move it into the range since it is not used elsewhere that I can see:
```
for _, elem in range strings.Split(str, ";") {
```

Reply:
Done